### ISSUE TITLE
MOSEI Data missmatch 

### ISSUE THREAD

**Issue by zsun227** [A2Zadeh/CMU-MultimodalSDK#98]
Hi, thanks for reading this issue.  

Basically speaking, I used two ways to get the MOSEI data.  

First I just downloaded the proprocessed data from the link http://immortal.multicomp.cs.cmu.edu/raw_datasets/old_processed_data/. 
And here is each dataset's size :  (train: 15290,test: 4832,  val: 2291) 

Then I used the newest version CMU-MultimodalSDK and downloaded the MOSEI dataset. I used the cmu_mosei_std_folds.py to seprate the whole dataset to train, test and val. Now each dataset's size is : (16331, 4662, 1871)

However, I know that you guys may have changed the MOSEI dataset. But the point is, test MAE results varied a lot when I used the same model (say LSTM) to process with those two dataset.  (Actually, for the newest MOSEI version, I'm a little bit worried about the distributions of valid set and test set, because I got a 0.77 MAE on valid testset but got a 0.84 on the testset. But when used the proprocessed dataset, the difference between testerror and valid error is fairly small)

However, I guess we may need to use the newest MOSEI dataset in our experiment right? 

Please tell me if my data separating part is wrong....

Thank you!

**Comment by A2Zadeh** [comment]
Hello @zsun227,

Yes, we did change the MOSEI dataset a bit after the original publication. The newest results are aligned with the workshop in ACL 2018 (you can find the proceedings [here](http://multicomp.cs.cmu.edu/acl2018multimodalchallenge/). 

As for your question, the discrepancy can be somewhat expected based on what model you are using. Normally the simpler models show the same performance between train/valid/test but more complex models can overfit to the train set or validation set a bit. That being said, the value for 0.77 is also not fully optimized (currently state of the art models are achieving 0.58 by both us and participants of the challenge). Here are some things to check:

1) Whether or not your model is overfitting on the train and validation data. You can see if there is a large gap between your train and validation. If there is not a large gap there, then your model may not be doing well on the train set, therefore not surprising that test results are not doing well. If there is a large gap there, then your model may be overfitting, and again not generalizing to validation and test data. Training multimodal models are a bit tough :)

2) Whether or not removing speech pauses will be helpful for your case. There is lots of speech pauses in the transcripts and sometimes they can be overwhelming for more complex models. If you pull the SDK again, I recently changed the word vectors to not include pauses, and there is a new field for download which does include pauses.

3) You could optimize for MAE directly. 

You should always run the newest version of the SDK and newest version of the features. We always keep them updated and make sure that all the issues with the dataset are corrected. 

Hope this helps.

**Comment by zsun227** [comment]
> Hello @zsun227,
> 
> Yes, we did change the MOSEI dataset a bit after the original publication. The newest results are aligned with the workshop in ACL 2018 (you can find the proceedings [here](http://multicomp.cs.cmu.edu/acl2018multimodalchallenge/)).
> 
> As for your question, the discrepancy can be somewhat expected based on what model you are using. Normally the simpler models show the same performance between train/valid/test but more complex models can overfit to the train set or validation set a bit. That being said, the value for 0.77 is also not fully optimized (currently state of the art models are achieving around 0.54 for validation which translates to 0.58 for test - by both us and participants of the challenge). Here are some things to check:
> 
> 1. Whether or not your model is overfitting on the train and validation data. You can see if there is a large gap between your train and validation. If there is not a large gap there, then your model may not be doing well on the train set, therefore not surprising that test results are not doing well. If there is a large gap there, then your model may be overfitting, and again not generalizing to validation and test data. Training multimodal models are a bit tough :)
> 2. Whether or not removing speech pauses will be helpful for your case. There is lots of speech pauses in the transcripts and sometimes they can be overwhelming for more complex models. If you pull the SDK again, I recently changed the word vectors to not include pauses, and there is a new field for download which does include pauses.
> 3. You could optimize for MAE directly. You should also make sure to run extensive hyperparameters search (>1000 combinations in general for complex models).
> 
> You should always run the newest version of the SDK and newest version of the features. We always keep them updated and make sure that all the issues with the dataset are corrected.
> 
> Hope this helps.

Thank you! This really helps
